Library/include/CSP/Systems/Users/Authentication.h

```diff
@@ -120,66 +123,28 @@ class CSP_API LoginStateResult : public csp::systems::ResultBase
 };
 
 
-/// @brief Result structure for a logout state request.
-class CSP_API LogoutResult : public NullResult
-{
-	/** @cond DO_NOT_DOCUMENT */
-	CSP_START_IGNORE
-	template <typename T, typename U, typename V, typename W> friend class csp::services::ApiResponseHandler;
-	friend class UserSystem;
-	CSP_END_IGNORE
-	/** @endcond */
-
-private:
-	LogoutResult();
-	LogoutResult(LoginState* InStatePtr);
-
-	void OnResponse(const csp::services::ApiResponseBase* ApiResponse) override;
-
-	LoginState* State;
-};
-
-
 /// @ingroup User System
 /// @brief @brief Data class used to contain information when the login token has changed
-class CSP_API LoginTokenReceived : public csp::systems::ResultBase
+class CSP_API LoginTokenInfoResult : public ResultBase
 {
 	/** @cond DO_NOT_DOCUMENT */
 	friend class UserSystem;
 	/** @endcond */
 
 public:
-	LoginTokenInfo& GetLoginTokenInfo();
+	[[nodiscard]]
 	const LoginTokenInfo& GetLoginTokenInfo() const;
 
 private:
-	LoginTokenReceived(void*) {};
-	LoginTokenReceived() {};
+	LoginTokenInfoResult() = default;
+	LoginTokenInfoResult(void*) {};
 
 	void FillLoginTokenInfo(const csp::common::String& AccessToken,
 							const csp::common::String& AuthTokenExpiry,
 							const csp::common::String& RefreshToken,
 							const csp::common::String& RefreshTokenExpiry);
 
-	LoginTokenInfo LoginTokenInfo;
-};
-
-
-/// @ingroup User System
-/// @brief @brief Data class used to contain information when a ping response is received
-class CSP_API PingResponseReceived : public csp::systems::ResultBase
```

Comment:
This type has been removed. All placed that previously used this type now use `NullResult`, as they are functionally the same, and the new type's name portrays to the developer what is being returned.